bittensor/_subtensor/subtensor_mock.py

```diff
@@ -104,10 +106,13 @@ def create_global_mock_process(self):
         try:
             operating_system = "OSX" if platform == "darwin" else "Linux"
             path = "./bin/chain/{}/node-subtensor".format(operating_system)
-            port = int(bittensor.__mock_entrypoints__[0].split(':')[1])
-            print(port)
+            ws_port = int(bittensor.__mock_entrypoints__[0].split(':')[1])
+            print(ws_port)
+            print(os.getpid())
+            baseport = get_random_unused_port()
+            rpc = get_random_unused_port()
```

Comment:
Turns out we only care about `ws_port`. Using `port+1` and `port+2` for the other two ports caused issues because the port in `mock_entrypoint` was generated by the PID. But often these processes are spawned sequentially, so the PIDs are as well, which means the subprocesses all squabble for the same port and everyone ends up losing.

Reply:
Nice catch!